main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -515,6 +514,37 @@ object Inliner {
     case _ => false
   }
 
+  /**
+    * Returns an inlined and beta-reduced version of `exp0.sym` if the [[shouldInlineDef]] predicate holds.
+    * Otherwise, returns an [[Expr.ApplyDef]] expression , where the subexpressions have been visited.
+    */
+  private def callSiteInlineDef(exp0: Expr.ApplyDef, ctx0: LocalContext)(implicit sym0: Symbol.DefnSym, sctx: SharedContext, root: OccurrenceAst.Root, flix: Flix): Expr = exp0 match {
+    case Expr.ApplyDef(sym, exps, _, _, _, loc) if shouldInlineDef(root.defs(sym), ctx0) =>
```

Comment:
This should be inlined up into the bigger pattern match and then you can also drop the guard and just use a plain old if.